### PR TITLE
added printer for SymmetricDifference

### PR DESCRIPTION
**Pull request by kaichogami**
Added `latex` and `unicode` printing for `SymmetricDifference`.

### CODE REVIEW

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -210,7 +210,8 @@ def symmetric_difference(self, other):
          return SymmetricDifference(self, other)
 
     def _symmetric_difference(self, other):
-        return Union(Complement(self, other), Complement(other, self))
+        if isinstance(other, FiniteSet) or isinstance(other, Interval):
```
Comment (hargup) [inline sympy/sets/sets.py]: Why this change? Complement and Union can work for many other different classes of sets.

File: sympy/printing/str.py [hunk 2]
```diff
@@ -645,6 +645,9 @@ def _print_Uniform(self, expr):
     def _print_Union(self, expr):
         return ' U '.join(self._print(set) for set in expr.args)
 
+    def _print_SymmetricDifference(self, expr):
+        return ' ^ '.join(self._print(set) for set in expr.args)
```
Comment (hargup) [inline sympy/printing/str.py]: Use the `∆` unicode symbol for unicode printing. Also add latex printer, use can use `\triangle` for the symbol.

File: sympy/printing/pretty/pretty.py [hunk 3]
```diff
@@ -1471,6 +1471,12 @@ def _print_Union(self, u):
         return self._print_seq(u.args, None, None, union_delimiter,
              parenthesize=lambda set: set.is_ProductSet or set.is_Intersection)
 
+    def _print_SymmetricDifference(self, u):
+
+        sym_delimeter = ' %s ' % pretty_atom('SymmetricDifference')
+
+        return self._print_seq(u.args, None, None, sym_delimeter)
```
Comment (kaichogami) [inline sympy/printing/pretty/pretty.py]: @hargup I didn't understand how `parenthesize` is used, so I haven't added anything there.

File: sympy/printing/pretty/pretty_symbology.py [hunk 4]
```diff
@@ -462,6 +462,7 @@ def xsym(sym):
     'Integers':         U('DOUBLE-STRUCK CAPITAL Z'),
     'Reals':            U('DOUBLE-STRUCK CAPITAL R'),
     'Union':            U('UNION'),
+    'SymmetricDifference':    U('INCREMENT'),
     'Intersection':     U('INTERSECTION'),
     'Ring':             U('RING OPERATOR')
 }
```
Comment (kaichogami) [inline sympy/printing/pretty/pretty_symbology.py]: Perhaps I should increase width between column?

File: sympy/sets/sets.py (hunk 1 above)
Comment (kaichogami) [inline sympy/sets/sets.py]: Alright, reverted back. `reduce` handles everything.

File: sympy/sets/sets.py [hunk 5]
```diff
@@ -210,7 +210,7 @@ def symmetric_difference(self, other):
          return SymmetricDifference(self, other)
 
     def _symmetric_difference(self, other):
-        return Union(Complement(self, other), Complement(other, self))
+         return Union(Complement(self, other), Complement(other, self))
```
Comment (hargup) [inline sympy/sets/sets.py]: Indentation is not a multiple of four. But I wonder didn't the test fail?

File: sympy/printing/pretty/pretty_symbology.py (hunk 4 above)
Comment (hargup) [inline sympy/printing/pretty/pretty_symbology.py]: This shouldn't be consequential, so do it if you feel like.

File: sympy/printing/str.py (hunk 2 above)
Comment (hargup) [inline sympy/printing/str.py]: You should remove this function. `^` is not the right symbol for SymmetricDifference. Without any specialized function the result should be something like
```
In [2]: str(SymmetricDifference(Interval(0, 1), Interval(3, 4), evaluate=False))
Out[2]: 'SymmetricDifference([0, 1], [3, 4])'
```

**Comment by hargup** [comment]
You should also add tests.

**Comment by kaichogami** [comment]
@hargup added printer for `unicode` and `latex`. Also added test for `latex`.

File: sympy/printing/tests/test_latex.py [hunk 6]
```diff
@@ -504,6 +504,13 @@ def test_latex_union():
         r"\left\{1, 2\right\} \cup \left[3, 4\right]"
 
 
+def test_latex_symmetric_difference():
+    assert latex(SymmetricDifference(Interval(2, 5), Interval(3, 5))) == \
+        r"\left[2, 3\right)"
+    assert latex(SymmetricDifference(Interval(1,10), Interval(2,20))) == \
```
Comment (hargup) [inline sympy/printing/tests/test_latex.py]: This `SymmetricDifference` evaluates to union of two intervals, you have to test if unevaluated `SymmetricDifference`s are being printed correctly.

File: sympy/printing/pretty/pretty_symbology.py [hunk 7]
```diff
@@ -449,21 +449,22 @@ def xsym(sym):
...
-    'ImaginaryUnit':    U('DOUBLE-STRUCK ITALIC SMALL I'),
-    'EmptySet':         U('EMPTY SET'),
-    'Naturals':         U('DOUBLE-STRUCK CAPITAL N'),
-    'Integers':         U('DOUBLE-STRUCK CAPITAL Z'),
-    'Reals':            U('DOUBLE-STRUCK CAPITAL R'),
-    'Union':            U('UNION'),
-    'Intersection':     U('INTERSECTION'),
-    'Ring':             U('RING OPERATOR')
+    # class                    how-to-display
+    'Exp1':                    U('SCRIPT SMALL E'),
```
Comment (hargup) [inline sympy/printing/pretty/pretty_symbology.py]: I'm not sure about the role `atoms_table` play in pretty printing. Try reverting these changes and see if you still get what you wanted to get? I didn't have to touch this dict while adding printer for the Set Difference class. See https://github.com/hargup/sympy/commit/324aa9a5ec016dd1cb5c2e1e6b94aff27036049c

**Comment by hargup** [comment]
The PR looks good but you also need to add tests for ascii and unicode pretty printing. You can have a look at the Set Difference PR for reference https://github.com/sympy/sympy/pull/7462

File: sympy/printing/pretty/pretty_symbology.py (hunk 7 above)
Comment (kaichogami) [inline sympy/printing/pretty/pretty_symbology.py]: @hargup `atoms_table` is a dict for accessing unicode symbols from the `unicodedata` module. Here the key `SymmetricDifference` is used in `pretty.py` module to print the `INCREMENT` unicode symbol for `SymmetricDifference` class. 
In case of `Complement` class you created, there was no need for any `unicode` symbol as `\` was used for `unicode` and `ascii` printing, which can be directly used.

File: sympy/printing/tests/test_latex.py [hunk 8]
```diff
@@ -504,6 +504,15 @@ def test_latex_union():
         r"\left\{1, 2\right\} \cup \left[3, 4\right]"
 
 
+def test_latex_symmetric_difference():
+    assert latex(SymmetricDifference(Interval(2, 5), Interval(3, 5))) == \
```
Comment (hargup) [inline sympy/printing/tests/test_latex.py]: Only the test with `evaluate=False` should be enough. You should remove other two tests.

File: sympy/printing/pretty/tests/test_pretty.py [hunk 9]
```diff
@@ -4830,6 +4830,12 @@ def test_pretty_Complement():
     assert upretty(S.Reals - S.Naturals) == u('ℝ \ ℕ')
 
 
+def test_pretty_SymmetricDifference():
```
Comment (hargup) [inline sympy/printing/pretty/tests/test_pretty.py]: Also add a test for ascii printing.

**Comment by kaichogami** [comment]
@hargup Added failing test for `ascii` printing. 
Do you think there is an appropriate `ascii` symbol that could be used for `SymmetricDifference`?

**Comment by hargup** [comment]
Sorry for the late response, the pr looks good. I'll merge it in 24h if there is no objection.

**Comment by hargup** [comment]
Thanks, it is in.
